python/dgl/distributed/partition.py

```diff
@@ -444,7 +471,7 @@ def load_partition_feats(
     return node_feats, edge_feats
 
 
-def load_partition_book(part_config, part_id):
+def load_partition_book(part_config, part_id, part_metadata=None):
```

Comment:
this function is a public API, please do not change it. try to figure out another way to achieve what you want.

Reply:
one alternative is wrap below logic with a new funtion `_load_xxx(metadata)`, then call it.

Reply:
I added a new function `_load_partition_book_from_metadata` to keep the `load_partition_book` unchanged